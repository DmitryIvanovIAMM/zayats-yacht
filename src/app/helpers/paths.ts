/* eslint-disable no-unused-vars */

export enum PATHS {
  landing = "/",
  quoteRequest = "/quote-request",
  gallery = "/gallery",
  services = "/services",
  instructions = "/instructions",
  privacyPolicy = "/privacy-policy",
  authCallback = "/auth-callback",
  scheduleManagement = "/schedule-management",
  destination = "/destination/:id",
}
