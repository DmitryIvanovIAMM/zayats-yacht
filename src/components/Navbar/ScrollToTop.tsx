import { Fade, useScrollTrigger } from '@mui/material';
import Box from '@mui/material/Box';
import React, { ReactNode } from 'react';

export interface ScrollTopProps {
  children: ReactNode;
}

const ScrollToTop: React.FC<ScrollTopProps> = ({ children }) => {
  // Note that you normally won't need to set the window ref as useScrollTrigger
  // will default to window.
  // This is only being set here because the demo is in an iframe.
  const trigger = useScrollTrigger({
    target: undefined,
    disableHysteresis: true,
    threshold: 100
  });

  const handleClick = () => {
    window.scrollTo({ top: 0, behavior: 'smooth' });
  };

  return (
    <Fade in={trigger}>
      <Box
        onClick={handleClick}
        role="presentation"
        sx={{ position: 'fixed', bottom: 11, right: 16 }}
      >
        {children}
      </Box>
    </Fade>
  );
};

export default ScrollToTop;
