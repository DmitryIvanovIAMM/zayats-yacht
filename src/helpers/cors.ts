// src/helpers/cors.ts

export function getAllowedOrigin(request: Request): string {
  const origin = request.headers.get('origin');
  return origin && /^https?:\/\/localhost(:\d+)?$/.test(origin) ? origin : '';
}

export function getCORSHeaders(request: Request): HeadersInit {
  const allowedOrigin = getAllowedOrigin(request);
  const headers: Record<string, string> = {
    'Content-Type': 'application/json'
  };
  if (allowedOrigin) {
    headers['Access-Control-Allow-Origin'] = allowedOrigin;
    headers['Access-Control-Allow-Credentials'] = 'true';
    headers['Access-Control-Allow-Methods'] = 'GET, OPTIONS, HEAD, PUT, POST, DELETE';
    headers['Access-Control-Allow-Headers'] =
      'X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version';
  }
  return headers;
}
