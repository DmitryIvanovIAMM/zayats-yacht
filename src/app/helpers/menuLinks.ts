import { PATHS } from '../helpers/paths';

export interface MenuLink {
  label: string;
  sectionName?: string;
  link?: string;
  scrollDuration?: number;
}

export const menuLinks = [
  {
    label: 'Schedule',
    sectionName: 'schedulesSection',
    scrollDuration: 500
  },
  {
    label: 'Get a Quote',
    link: PATHS.quoteRequest
  },
  {
    label: 'Gallery',
    link: PATHS.gallery
  },
  {
    label: 'Services',
    link: PATHS.services
  },
  {
    label: 'Instructions',
    link: PATHS.instructions
  },
  {
    label: 'Testimonials',
    sectionName: 'testimonials',
    scrollDuration: 500
  },
  {
    label: 'About Us',
    sectionName: 'aboutUsSection',
    scrollDuration: 1000
  },
  {
    label: 'Contact Us',
    sectionName: 'contactsSection',
    scrollDuration: 1500
  }
];

export const editScheduleMenuLinks = [
  {
    label: 'Home',
    link: '/',
    sectionName: 'quoteRequestSection',
    scrollDuration: 500
  }
];
