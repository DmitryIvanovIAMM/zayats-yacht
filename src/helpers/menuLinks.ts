import { PATHS } from './paths';
import { Roles } from '@/utils/types';

export interface MenuLink {
  label: string;
  section?: string;
  link?: string;
  scrollDuration?: number;
}

export const baseMenuLinks = [
  {
    label: 'Schedule',
    link: PATHS.landing,
    section: 'schedule-section',
    scrollDuration: 500
  },
  {
    label: 'Get a Quote',
    link: PATHS.quoteRequest
  },
  {
    label: 'Gallery',
    link: PATHS.gallery
  },
  {
    label: 'Services',
    link: PATHS.services
  },
  {
    label: 'Instructions',
    link: PATHS.instructions
  },
  {
    label: 'Testimonials',
    link: PATHS.landing,
    section: 'testimonials-section',
    scrollDuration: 500
  },
  {
    label: 'About Us',
    link: PATHS.landing,
    section: 'about-us-section',
    scrollDuration: 1000
  },
  {
    label: 'Contact Us',
    link: PATHS.landing,
    section: 'contact-us-section',
    scrollDuration: 1500
  }
];

export const adminSectionMenuLink = {
  label: 'Admin',
  link: PATHS.admin
};

export const adminMenuLinks = [adminSectionMenuLink, ...baseMenuLinks];

export const getMenuLinksForRole = (role: Roles = Roles.User) => {
  if (role === Roles.Admin) {
    return adminMenuLinks;
  } else {
    return baseMenuLinks;
  }
};
