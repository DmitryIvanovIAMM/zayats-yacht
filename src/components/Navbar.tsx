"use client";

import * as React from "react";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";
import IconButton from "@mui/material/IconButton";
import Typography from "@mui/material/Typography";
import MenuIcon from "@mui/icons-material/Menu";
import Button from "@mui/material/Button";
import Image from "next/image";
import logoImage from "../assets/images/allied_yacht_vertical_png_120.png";
import {
  Divider,
  Drawer,
  List,
  ListItem,
  ListItemButton,
  ListItemText,
  Slide,
  useScrollTrigger,
} from "@mui/material";
import { menuLinks } from "@/app/helpers/menuLinks";

const drawerWidth = 240;

export interface NavbarProps {
  isAuthenticated: boolean;
  children?: React.ReactElement<unknown>;
}

function HideOnScroll(props: NavbarProps) {
  const { children } = props;
  // Note that you normally won't need to set the window ref as useScrollTrigger
  // will default to window.
  // This is only being set here because the demo is in an iframe.
  const trigger = useScrollTrigger({
    target: undefined,
  });

  return (
    <Slide appear={false} direction="down" in={!trigger}>
      {children ?? <div />}
    </Slide>
  );
}

const Navbar: React.FC<NavbarProps> = (props) => {
  const { isAuthenticated } = props;
  console.log("Navbar().  isAuthenticated: ", isAuthenticated);
  const [mobileOpen, setMobileOpen] = React.useState(false);

  const handleDrawerToggle = () => {
    setMobileOpen((prevState) => !prevState);
  };

  const drawer = (
    <Box
      onClick={handleDrawerToggle}
      sx={{
        textAlign: "center",
        paddingLeft: "10px",
        paddingRight: "10px",
        backgroundColor: "#0A2A3B",
      }}
    >
      <Typography
        variant="h6"
        sx={{
          my: 2,
          textAlign: "center",
          color: "white",
          fontWeight: "600",
          fontSize: "14px",
          textTransform: "uppercase",
          //fontFamily: "Montserrat",
        }}
      >
        Zayats-Yacht Transport
      </Typography>
      <hr
        style={{
          border: "none",
          color: "#00b39e",
          backgroundColor: "#00b39e",
          height: "1px",
          marginTop: "35px",
        }}
      ></hr>
      <Divider />
      <List>
        {menuLinks.map((item) => (
          <div key={item.label}>
            <ListItem disablePadding>
              <ListItemButton
                sx={{
                  textAlign: "left",
                  color: "white",
                  fontWeight: "600",
                  fontSize: "14px",
                  textTransform: "uppercase",
                  //fontFamily: "Montserrat",
                }}
              >
                <ListItemText primary={item.label} />
              </ListItemButton>
            </ListItem>
            <hr
              style={{
                border: "none",
                color: "#00b39e",
                backgroundColor: "#00b39e",
                height: "1px",
              }}
            ></hr>
          </div>
        ))}
      </List>
    </Box>
  );
  const container = undefined;

  return (
    <>
      <HideOnScroll {...props}>
        <AppBar sx={{ backgroundColor: "white", color: "blue" }}>
          <Toolbar>
            <Box sx={{ flexGrow: 1 }}>
              <IconButton
                size="large"
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={handleDrawerToggle}
                color="inherit"
              >
                <MenuIcon />
              </IconButton>
            </Box>
            <Box sx={{ flexGrow: 1 }}>
              <Image
                src={logoImage}
                width={105} // 81 125
                height={71} //58 90*/
                alt="Allied-Yat logo"
                priority={true}
              />
            </Box>

            <Box sx={{ flexGrow: 0 }}>
              <Button>Sign In</Button>
            </Box>
          </Toolbar>
        </AppBar>
      </HideOnScroll>
      <nav>
        <Drawer
          container={container}
          variant="temporary"
          open={mobileOpen}
          onClose={handleDrawerToggle}
          ModalProps={{
            keepMounted: true, // Better open performance on mobile.
          }}
          sx={{
            display: { xs: "block" },
            "& .MuiDrawer-paper": {
              boxSizing: "border-box",
              width: drawerWidth,
              height: "auto",
            },
            opacity: 0.9,
          }}
        >
          {drawer}
        </Drawer>
      </nav>
    </>
  );
};
export default Navbar;
