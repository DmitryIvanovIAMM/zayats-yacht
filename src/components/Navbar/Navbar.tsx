'use client';

import React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import MenuIcon from '@mui/icons-material/Menu';
import Button from '@mui/material/Button';
import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
import Image from 'next/image';
import logoImage from '../../../public/images/allied_yacht_vertical_png_120.png';
import {
  CssBaseline,
  Divider,
  Drawer,
  Fab,
  List,
  ListItem,
  ListItemButton,
  ListItemText,
  Slide,
  useScrollTrigger
} from '@mui/material';
import { menuLinks } from '@/app/helpers/menuLinks';
import ScrollToTop from './ScrollToTop';
import { secondary } from '@/components/colors';

const drawerWidth = 240;
const leftNavigationSx = {
  textAlign: 'center',
  paddingLeft: '10px',
  paddingRight: '10px',
  backgroundColor: '#0A2A3B'
};
const dividerStyle = {
  border: 'none',
  color: secondary.main,
  backgroundColor: secondary.main,
  height: '1px'
};
const menuItemSx = {
  textAlign: 'left',
  color: 'white',
  fontWeight: '600',
  fontSize: '14px',
  textTransform: 'uppercase',
  paddingLeft: '0px',
  paddingRight: '0px'
};

export interface NavbarProps {
  isAuthenticated: boolean;
  children?: React.ReactElement<unknown>;
}

function HideOnScroll(props: NavbarProps) {
  const { children } = props;
  // Note that you normally won't need to set the window ref as useScrollTrigger
  // will default to window.
  // This is only being set here because the demo is in an iframe.
  const trigger = useScrollTrigger({
    target: undefined
  });

  return (
    <Slide appear={false} direction="down" in={!trigger}>
      {children ?? <div />}
    </Slide>
  );
}

const Navbar: React.FC<NavbarProps> = (props) => {
  const { isAuthenticated } = props;
  const [menuOpen, setMenuOpen] = React.useState(false);

  const handleDrawerToggle = () => {
    setMenuOpen((prevState) => !prevState);
  };

  const drawer = (
    <Box onClick={handleDrawerToggle} sx={leftNavigationSx}>
      <div>
        <Typography variant="h6" sx={{ ...menuItemSx, my: 2, textAlign: 'center' }}>
          Zayats-Yacht Transport
        </Typography>
        <Image src={logoImage} width={52} height={36} alt="Allied-Yacht logo" priority={true} />
      </div>
      <hr style={{ ...dividerStyle, marginTop: '10px' }}></hr>
      <Divider />
      <List>
        {menuLinks.map((item) => (
          <div key={item.label}>
            <ListItem disablePadding>
              <ListItemButton sx={menuItemSx}>
                <ListItemText primary={item.label} />
              </ListItemButton>
            </ListItem>
            <hr style={dividerStyle}></hr>
          </div>
        ))}
      </List>
    </Box>
  );
  const container = undefined;

  return (
    <div>
      <CssBaseline />
      <HideOnScroll {...props}>
        <AppBar sx={{ backgroundColor: 'white', color: 'secondary.dark' }}>
          <Toolbar disableGutters sx={{ padding: '0 10px 0 0px' }}>
            <Box sx={{ flexGrow: 1 }}>
              <IconButton
                size="large"
                aria-label="account of current user"
                aria-controls="menu-appbar"
                aria-haspopup="true"
                onClick={handleDrawerToggle}
                color="inherit"
                data-testid="left-menu-button"
              >
                <MenuIcon />
              </IconButton>
              <Button
                variant={'contained'}
                sx={{ backgroundColor: 'secondary.dark' }}
                size={'small'}
              >
                Get Quote
              </Button>
            </Box>
            <Box
              sx={{
                flexGrow: 1,
                marginTop: '4px'
              }}
            >
              <Image
                src={logoImage}
                width={105} // 81 125
                height={71} //58 90*/
                alt="Allied-Yacht logo"
                priority={true}
              />
            </Box>

            <Box sx={{ flexGrow: 0 }}>
              <Button
                variant={'contained'}
                sx={{ backgroundColor: 'secondary.dark' }}
                size={'small'}
              >
                {isAuthenticated ? 'LogOut' : 'SignIn'}
              </Button>
            </Box>
          </Toolbar>
        </AppBar>
      </HideOnScroll>
      <nav>
        <Drawer
          container={container}
          variant="temporary"
          open={menuOpen}
          onClose={handleDrawerToggle}
          ModalProps={{
            keepMounted: true // Better open performance on mobile.
          }}
          sx={{
            display: { xs: 'block' },
            '& .MuiDrawer-paper': {
              boxSizing: 'border-box',
              width: drawerWidth,
              backgroundColor: '#0A2A3B'

              //height: 'auto'
            },
            opacity: 0.9
          }}
        >
          {drawer}
        </Drawer>
      </nav>
      <ScrollToTop>
        <Fab size="small" aria-label="scroll back to top">
          <KeyboardArrowUpIcon />
        </Fab>
      </ScrollToTop>
    </div>
  );
};
export default Navbar;
