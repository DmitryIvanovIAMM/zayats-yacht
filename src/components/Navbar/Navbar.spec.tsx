import { act, render } from '@testing-library/react';
import Navbar, { NavbarProps } from '@/components/Navbar/Navbar';
import { menuLinks } from '@/app/helpers/menuLinks';

const setup = async (propsOverride?: Partial<NavbarProps>) => {
  const props = {
    isAuthenticated: false,
    ...propsOverride
  };
  const container = render(<Navbar {...props} />);
  return {
    container
  };
};

describe('Navbar component', () => {
  it('should render correctly', async () => {
    const { container } = await setup({ isAuthenticated: true });

    expect(container).toMatchSnapshot();
  });

  it('should show logo', async () => {
    const { container } = await setup({ isAuthenticated: true });

    expect(container.getByAltText('Allied-Yacht logo')).toBeInTheDocument();
  });

  it('should show Sign In button for non authenticated user', async () => {
    const { container } = await setup({ isAuthenticated: true });

    expect(container.getByText('Sign In')).toBeInTheDocument();
  });

  it('should contains left navigation menu items by default', async () => {
    const { container } = await setup({ isAuthenticated: true });

    act(() => {
      // no click required in current realization because drawer always mounted
      container.getByTestId('left-menu-button').click();
    });
    menuLinks.forEach((menuLink) => {
      expect(container.getByText(menuLink.label)).toBeInTheDocument();
    });
  });
});
