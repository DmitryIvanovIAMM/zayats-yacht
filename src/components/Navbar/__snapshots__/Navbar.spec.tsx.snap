// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <header
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-12db59r-MuiPaper-root-MuiAppBar-root"
          style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
        >
          <div
            class="MuiToolbar-root MuiToolbar-regular css-1uj5ifc-MuiToolbar-root"
          >
            <div
              class="MuiBox-root css-i9gxme"
            >
              <button
                aria-controls="menu-appbar"
                aria-haspopup="true"
                aria-label="account of current user"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1vipebp-MuiButtonBase-root-MuiIconButton-root"
                data-testid="left-menu-button"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                  data-testid="MenuIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                  />
                </svg>
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-1ccw0ge-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Get Quote
              </button>
            </div>
            <div
              class="MuiBox-root css-164ig2l"
            >
              <img
                alt="Allied-Yacht logo"
                data-nimg="1"
                decoding="async"
                fetchpriority="high"
                height="71"
                src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
                style="color: transparent;"
                width="105"
              />
            </div>
            <div
              class="MuiBox-root css-2uchni"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-1ccw0ge-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                LogOut
              </button>
            </div>
          </div>
        </header>
        <nav />
        <div
          class="MuiBox-root css-omdpw6"
          role="presentation"
          style="opacity: 0; visibility: hidden;"
        >
          <button
            aria-label="scroll back to top"
            class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default css-14bntoc-MuiButtonBase-root-MuiFab-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
              data-testid="KeyboardArrowUpIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiDrawer-root MuiDrawer-modal MuiModal-root MuiModal-hidden css-zw5q8m-MuiModal-root-MuiDrawer-root"
      role="presentation"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root MuiModal-backdrop css-4nmryk-MuiBackdrop-root-MuiModal-backdrop"
        style="opacity: 0; visibility: hidden;"
      />
      <div
        data-testid="sentinelStart"
        tabindex="-1"
      />
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation16 MuiDrawer-paper MuiDrawer-paperAnchorLeft css-cyqh1t-MuiPaper-root-MuiDrawer-paper"
        style="--Paper-shadow: 0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12); visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
        tabindex="-1"
      >
        <div
          class="MuiBox-root css-2t7qt4"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-j4hmdc-MuiTypography-root"
          >
            Zayats-Yacht Transport
          </h6>
          <hr
            style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px; margin-top: 35px;"
          />
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1jpc804-MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding css-b7dj7n-MuiList-root"
          >
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      Schedule
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      Get a Quote
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      Gallery
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      Services
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      Instructions
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      Testimonials
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      About Us
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
            <div>
              <li
                class="MuiListItem-root MuiListItem-gutters css-19f6boz-MuiListItem-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1vra68c-MuiButtonBase-root-MuiListItemButton-root"
                  role="button"
                  tabindex="0"
                >
                  <div
                    class="MuiListItemText-root css-cfq8qh-MuiListItemText-root"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-rizt0-MuiTypography-root"
                    >
                      Contact Us
                    </span>
                  </div>
                </div>
              </li>
              <hr
                style="color: rgb(51, 153, 102); background-color: rgb(51, 153, 102); height: 1px;"
              />
            </div>
          </ul>
        </div>
      </div>
      <div
        data-testid="sentinelEnd"
        tabindex="-1"
      />
    </div>
  </body>,
  "container": <div>
    <div>
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-12db59r-MuiPaper-root-MuiAppBar-root"
        style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular css-1uj5ifc-MuiToolbar-root"
        >
          <div
            class="MuiBox-root css-i9gxme"
          >
            <button
              aria-controls="menu-appbar"
              aria-haspopup="true"
              aria-label="account of current user"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1vipebp-MuiButtonBase-root-MuiIconButton-root"
              data-testid="left-menu-button"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="MenuIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M3 18h18v-2H3zm0-5h18v-2H3zm0-7v2h18V6z"
                />
              </svg>
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-1ccw0ge-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Get Quote
            </button>
          </div>
          <div
            class="MuiBox-root css-164ig2l"
          >
            <img
              alt="Allied-Yacht logo"
              data-nimg="1"
              decoding="async"
              fetchpriority="high"
              height="71"
              src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=128&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
              style="color: transparent;"
              width="105"
            />
          </div>
          <div
            class="MuiBox-root css-2uchni"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-1ccw0ge-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              LogOut
            </button>
          </div>
        </div>
      </header>
      <nav />
      <div
        class="MuiBox-root css-omdpw6"
        role="presentation"
        style="opacity: 0; visibility: hidden;"
      >
        <button
          aria-label="scroll back to top"
          class="MuiButtonBase-root MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default MuiFab-root MuiFab-circular MuiFab-sizeSmall MuiFab-default css-14bntoc-MuiButtonBase-root-MuiFab-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="KeyboardArrowUpIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7.41 15.41 12 10.83l4.59 4.58L18 14l-6-6-6 6z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
